src/client.ts

```diff
@@ -1051,6 +1063,21 @@ export class AwaitConnection {
     this.connected = false;
   }
 
+  isClosed(): boolean {
+    return !this.connected;
+  }
+
+  /** @internal */
+  setProxy(user: ConnectionProxy | null): void {
```

Comment:
Can we also prefix it with an `_`? This is *really* internal (and we'll prefix private attributes with `_`s in all public classes later anyways).